src/wp-includes/plugin.php

```diff
@@ -39,6 +39,38 @@
 if ( ! isset( $wp_current_filter ) )
 	$wp_current_filter = array();
 
+/**
+ * Check that the callback parameters match the number passed to add_action() or add_filter().
+ *
+ * @since 1.3.0
+ *
+ * @param $function_name   add_action() or add_filter(), as returned by __FUNCTION__.
+ * @param $function_to_add The callback to check.
+ * @param $accepted_args   The number of arguments claimed.
+ */
+function _check_callback_params($function_name, $function_to_add, $accepted_args)
+{
+    if ( defined( 'CP_STRICT' ) && CP_STRICT ) {
+        if ( is_string( $function_to_add ) && function_exists( $function_to_add ) ) {
+            $rf = new ReflectionFunction( $function_to_add );
```

Comment:
Instead of *reflection* is there any other alternatives we could use here? My concern is that reflection I assume creates duplicate objects rather than references - does this double the memory footprint of the objects? Would we need to garbage collect in some way here?

Reply:
I'm not aware of another way to do this - if someone knows of one I'd love to know about it.

I don't think memory is an issue here; this is a setting that should only ever be used by developers and never in production, so whatever machinations PHP goes through internally shouldn't be important.

Reply:
No I'm not aware of any other method either off the top of my head :(

Also I think even with the fact this is to be run in development we should still consider the memory footprint. If we don't have to double the initial filter memory use for a *working* dev instance that would be better.

Reply:
That's partly why I said this: https://github.com/ClassicPress/ClassicPress/pull/502#issuecomment-536658490

Reply:
Ahh I see, I guess that comment came in while I was reviewing this cos I didn't see it till just now lol

Reply:
I think reflection is more of a speed issue than a memory issue. From what I remember, reflection objects are just pointers to the "real" objects, but the runtime does need to introspect the "real" object in order to fill the reflection object with data, which is a bit slow.